core/lib/env_config/src/contracts.rs

```diff
@@ -4,12 +4,22 @@ use crate::{envy_load, FromEnv};
 
 impl FromEnv for ContractsConfig {
     fn from_env() -> anyhow::Result<Self> {
-        envy_load("contracts", "CONTRACTS_")
+        let mut contracts: ContractsConfig = envy_load("contracts", "CONTRACTS_")?;
+        // Note: we are renaming the bridge, the address remains the same
```

Comment:
I'm confused about assumptions and invariants here.
We have 2 variables: l2_erc20 and l2_shared_bridge

And we want to say:
* at least one of them must be set in the config
* their values must be equal (if one is not set, then we should set the other to the value of the first)

The cases I'm missing:
* what if both are not set - I guess then both should be None, right?
* what if they are set to DIFFERENT values in the config? shoudl we allow it, or is it a config bug?